src/test/java/com/github/bibsysdev/nva/cristin/person/CristinPersonsHandlerTest.java

```diff
@@ -0,0 +1,227 @@
+import java.net.URISyntaxException;
+import java.nio.charset.StandardCharsets;
+import java.util.Map;
+import javax.ws.rs.core.HttpHeaders;
+import javax.ws.rs.core.MediaType;
+import no.unit.nva.testutils.HandlerRequestBuilder;
+import nva.commons.apigateway.ApiGatewayHandler;
+import nva.commons.apigateway.GatewayResponse;
+import nva.commons.core.Environment;
+import nva.commons.core.JsonUtils;
+import nva.commons.core.ioutils.IoUtils;
+import org.apache.commons.codec.Charsets;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.function.Executable;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ArgumentsSource;
+import org.mockito.Mockito;
+
+public class CristinPersonsHandlerTest {
+
+    private static final String LANGUAGE_KEY = "language";
+    private static final String EMPTY_STRING = "";
+    private static final String LANGUAGE_NB = "nb";
+    private static final String LANGUAGE_INVALID = "invalid";
+    private static final String NAME_MATTIAS = "Mattias";
+    private static final String INVALID_JSON = "This is not valid JSON!";
+    private static final String EMPTY_LIST_STRING = "[]";
+    private static final String ALLOW_ALL_ORIGIN = "*";
+    private static final String NVA_ONE_PERSON_JSON_FILE = "nva_one_person.json";
+    private static final String NVA_PERSONS_NON_ENRICHED_JSON_FILE = "nva_persons_non_enriched.json";
+    private static final String NVA_PERSONS_NO_HITS_JSON_FILE = "nva_persons_no_hits.json";
+    private static final String CRISTIN_ONE_PERSON_JSON_FILE = "cristin_one_person.json";
+    private final ObjectMapper objectMapper = JsonUtils.objectMapper;
+    private final Environment environment = new Environment();
+    private CristinApiClient cristinApiClientStub;
+    private Context context;
+    private ByteArrayOutputStream output;
+    private CristinPersonsHandler handler;
+
+    @Test
+    public void handlerReturnsOkWhenInputContainsName() throws Exception {
+        GatewayResponse<PersonsWrapper> response = sendDefaultQuery();
+        assertEquals(HttpURLConnection.HTTP_OK, response.getStatusCode());
+    }
+
+    @Test
+    public void handlerIgnoresErrorsWhenTryingToEnrichPersonInformation() throws Exception {
+        cristinApiClientStub = Mockito.spy(cristinApiClientStub);
+        doThrow(new IOException()).when(cristinApiClientStub).getPerson(any(), any());
+        handler = new CristinPersonsHandler(cristinApiClientStub, environment);
+        GatewayResponse<PersonsWrapper> response = sendDefaultQuery();
+
+        assertEquals(HttpURLConnection.HTTP_OK, response.getStatusCode());
+        assertEquals(MediaType.APPLICATION_JSON, response.getHeaders().get(HttpHeaders.CONTENT_TYPE));
+    }
+
+    @Test
+    public void handlerThrowsInternalErrorWhenQueryingPersonsFails() throws Exception {
+        cristinApiClientStub = Mockito.spy(cristinApiClientStub);
```

Comment:
I think spy is meant to be used on real objects, and sparingly if that. Wiremock probably a better solution.